services/sqs.cfc

```diff
@@ -28,19 +28,240 @@ component {
+            array messageAttriubes = [ ],
+            string messageDeduplicationId,
+            string messageGroupId
+    ) {
+        var requestSettings = api.resolveRequestSettings( argumentCollection = arguments );
+        var payload = { 'Action': 'SendMessage', 'MessageBody': message };
+        if( isDefined( 'delaySeconds' ) && isNumeric( delaySeconds ) ) {
+            structAppend( payload, { 'DelaySeconds': delaySeconds } );
+        }
+        for( var idx=1; idx LTE arrayLen(messageAttriubes); idx++ ) {
+            structAppend( payload, { 
+                'MessageAttribute.#idx#.Name': messageAttriubes[idx].Name,
+                'MessageAttribute.#idx#.Value.StringValue': messageAttriubes[idx].Value,
+                'MessageAttribute.#idx#.Value.DataType': messageAttriubes[idx].DataType
+            });
+        }
+        if( isDefined( 'messageDeduplicationId' ) && len( messageDeduplicationId ) ) {
+            structAppend( payload, { 'MessageDeduplicationId': messageDeduplicationId } );
+        }
+        if( isDefined( 'messageGroupId' ) && len( messageGroupId ) ) {
+            structAppend( payload, { 'MessageGroupId': messageGroupId } );
+        }
+        var apiResponse = apiCall(
+            requestSettings,
+            'GET',
+            '/' & queueName,
+            payload
+        );
+        if ( apiResponse.statusCode == 200 ) {
+            apiResponse[ 'data' ] = utils.parseXmlDocument( apiResponse.rawData );
+        }
+        return apiResponse;
+    }
+
+    /**
+    * Receives SQS messages
+    * https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_ReceiveMessage.html
+    * @queueName Required string. The name of the queue to receive messages from (e.g. "123456789/my-sqs-queue").
+    * @maxNumberOfMessages Optional numeric. The number of messages to receive from the queue. The mmaxNumberOfMessages is limited to 10.
+    * @visibilityTimeout Optional numeric. The amount of time until the message will be available in subsequent requests if not deleted.
+    * Note: When AWS receives more than one message, ReceiveMessageResult is an array. 
+    *       When only one message is received, ReceiveMessageResult it is not an array.
+    * @waitTimeSeconds Optional numeric. The duration (in seconds) for which the call waits for a message to arrive in the queue before returning.
+    * @attribueNames Optional array. An array of strings for attributes that need to be returned along with each message. ['All'] returns all attributes.
+    * @messageAttributeNames Optional array. An array of strings for user-defined attributes. Case-sensitive. ['All'] returns all user-defined attributes.
+    * Example: ['userAttribute1','userAttribute2']
+    */
+    public any function receiveMessage(
+		    required string queueName,
+            numeric maxNumberOfMessages = 1,
+            numeric visibilityTimeout,
+            numeric waitTimeSeconds,
+            array attribueNames = [ ],
```

Comment:
I believe this should be `attributeNames`. It is also typoed below.

Reply:
Fixed.